instrumentation/kafka/kafka-clients/kafka-clients-common/library/README.md

```diff
@@ -0,0 +1,233 @@
+# Kafka Metrics
+
+The Kafka client exposes metrics via `org.apache.kafka.common.metrics.MetricsReporter` interface.
+OpenTelemetry provides an implementation that bridges the metrics into OpenTelemetry.
+
+To use, configure `OpenTelemetryKafkaMetrics` with an OpenTelemetry instance
+via `OpenTelemetryKafkaMetrics#setOpenTelemetry(OpenTelemetry)`, and include a reference to this
+class in kafka producer or consumer configuration, i.e.:
+
+```java
+Map<String, Object> config = new HashMap<>();
+config.put(ProducerConfig.METRIC_REPORTER_CLASSES_CONFIG, OpenTelemetryKafkaMetrics.class.getName());
+config.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, kafka.getKafkaConnectString());
+...
+try (KafkaProducer<byte[], byte[]> producer = new KafkaProducer<>(config)) { ... }
+```
+
+Note: Kafka reports several metrics at multiple attribute granularities. For
+example, `records-consumed-total` is reported with attribute key `[client-id]`
+and `[client-id, topic]`. If you analyze the sum of records consumed, ignoring dimensions, backends
+are likely to double count. To alleviate this, `OpenTelemetryKafkaMetrics` detects this
+scenario and only records the most granular set of attributes available. In the case
+of `records-consumed-total`, it reports `[client-id, topic]` and ignores `[client-id]`.
+
+The following table shows the full set of metrics exposed by the kafka client, and the corresponding
+OpenTelemetry metric each maps to (if available). Empty values in the Instrument Name, Instrument
+Description, etc column indicates there is no registered mapping for the metric and data is NOT
+collected.
+
+| # | Metric Group | Metric Name | Attribute Keys | Instrument Name | Instrument Description | Instrument Type |
+|---|--------------|-------------|----------------|-----------------|------------------------|-----------------|
+| 1 | `app-info` | `commit-id` | `client-id` |  |  |  |
```

Comment:
what does the # mean?

Reply:
Just a row counter. Added to get some intuition about how many instruments I was dealing with. Happy to delete it.

Reply:
I think I'd delete it (or add explanation in the text since it looks like maybe it's a meaningful "id")